blog/2025-06-10-ducklake/index.mdx

```diff
@@ -0,0 +1,453 @@
+---
+slug: ducklake
+title: "Get your data ducks in a row with DuckLake"
+description: >
+  Combine SQL and object storage data seamlessly with DuckLake, a data lakehouse
+  solution that works with Tigris. Query big and small data without egress fees
+  and simplify your data analytics workflows.
+image: ./ducks-in-a-lake.jpg
+keywords: [object storage, blob storage, s3, big data, data lake, duckdb]
+authors: [xe, ks]
+tags: [cloud, big data, data lake, duckdb]
+---
+
+import InlineCta from "@site/src/components/InlineCta";
+import PullQuote from "@site/src/components/PullQuote";
+
+![](./ducks-in-a-lake.jpg)
+
+There’s a new data lake on the market: DuckLake. It’s worth the hype because it
+gives you an open standard that not only enables you to run queries on your data
+lakes from anywhere, it outright encourages you to run your query, metadata, and
+storage layers separately in whatever platform works best for you. DuckLake
+moves the metadata layer out of object storage and into a dedicated database.
+But why would an object storage company like Tigris be excited about putting
+_less_ data in object storage?
+
+Iceberg and Delta Lake both store their metadata in files on object storage:
+Delta Lake uses a translation log file within the object storage itself, whereas
+Iceberg uses an external catalog that can be stored in a DB or in object
+storage. The trouble comes when you handle concurrent writes. Someone has to
+“win” and object storage systems don’t let you lock files without coordination
+via DynamoDB or another service. DuckLake lets you handle concurrent writes via
+the concurrent write support of your database engine of choice, you get all of
+the benefits of a normal database without having to have all your data in one
+disk. You get all the advantages of a database by actually using a database!
+
+But if you really want to take advantage of that flexibility and choice that
+DuckLake offers, you need to unstick all the data in your lakehouse from
+$BigCloud’s storage where every gigabyte of egress to a tool in another cloud
+has a cost. The metadata layer being separate from the data also eliminates
+another point where other solutions can get your provider to dig you in the cost
+department. Storing all your actual data in Tigris is just the cherry on the top
+of the cost savings sundae.
```

Comment:
needs something close to what Ovais was explaining with why moving the metadata out of object storage makes it easier to scale duckdb and why you want object storage for your lake house